Source/dom/layers/StyledLayer.js

```diff
@@ -16,12 +16,20 @@ Factory.registerClass(StyledLayer, MSImmutableStyledLayer)
 StyledLayer.define('style', {
   get() {
     const style = Style.fromNative(this._object.style())
+    style._parentLayer = this
     return style
   },
   set(style) {
     if (this.isImmutable()) {
       return
     }
+    if (style && style._sharedStyle) {
+      this._object.setSharedStyleID(style._sharedStyle.id)
+    }
+    if (style && style.sharedStyleId) {
+      this._object.setSharedStyleID(style.sharedStyleId)
+    }
```

Comment:
It would probably be worth having a `sharedStyle` property on this object as well? I'm not sure `style._sharedStyle` would be 100% reliable so having that as a preferred way to set it would be good.

Reply:
so I added the ability to set a SharedStyle directly:
```js
layer.style = sharedStyle
```

I'm a bit reluctant to add a `sharedStyle` property as the sentence we want to enable is:
- "set the style of the layer to be the same as the shared style" -> `layer.style = sharedStyle`

Reply:
As a use case example that is a little confusing to me; Would it be possible to set a Style, associated with a SharedStyle without linking it? I would expect setting a Style would _not_ link, while setting a SharedStyle would.

Reply:
yes:
```
const style = sharedStyle.newInstance() // create a new style instance linked to the shared style
style.sharedStyle = undefined // unlink the style
layer.style = style
```

Reply:
could you write a small code example of what it would look like? I'm struggling to visualize it

Reply:
I think the tricky part to understand without knowing the internals is that, in a sense,  a Style can "represent" a SharedStyle (if it has a sharedStyleId).

Without realizing that I would expect the following
```javascript
const aSharedStyle = // a SharedStyle
layer.sharedStyle = aSharedStyle // sets and links the style
layer.style = aSharedStyle.style // just sets the style w/out linking
// and maybe
layer.sharedStyle = aSharedStyle.style // sets and links the style
```
In other words, to link you would have to interact with the sharedStyle property. But again, that doesn't make as much sense knowing the internals

A possible example of confusion right now:
```javascript
const styleA = layerA.style // not shared
const styleB = layerB.style // shared style
layer.style = styleA // just sets the style 
layer.style = styleB // sets and links the style
```
Most of the time this is probably the intended behavior which is why I say it seems like a good option, so long as it is well documented.

Reply:
ok, I resigned myself to limit the analogy to just SymbolMaster/SharedStyle. There won't be any analogy with SymbolInstance even tho there is a `SharedStyle.getAllInstances()` which returns Styles.

So now there is a `layer.sharedStyleId` property (and a convenient `layer.sharedStyle` as well). Setting it does NOT change the style of the layer. To sync the style of a layer with a shared style, you will need to do `layer.style.syncWithSharedStyle(layer.sharedStyle)`.

Reply:
Do **you** think this is better? I'm afraid I should have placed more emphasis on "Most of the time this is probably intended". I hate to say this after you've done the work but seeing the alternative, the "magic" approach does seem cleaner, you can set a Style (w/ or w/out a sharedID) or SharedStyle, and the style you set will define how it is attached.

I assume linking and applying is the more common goal, rather than just linking. Just linking could still be done by applying the sharedStyleId on the style though right?

Reply:
I’m not too sure either way. Some cases makes more sense before (as in the analogy between SymbolInstance and Style made some sense) but at the same time I think it’s more explicit now and explicit always win